internal/service/elbv2/target_group_test.go

```diff
@@ -1293,19 +1293,27 @@ func TestAccELBV2TargetGroup_Stickiness_invalidALB(t *testing.T) {
 		CheckDestroy:             testAccCheckTargetGroupDestroy(ctx),
 		Steps: []resource.TestStep{
 			{
-				Config:      testAccTargetGroupConfig_stickinessValidity(rName, "HTTP", "source_ip", true),
+				Config:      testAccTargetGroupConfig_stickinessValidity(rName, "HTTP", "source_ip", true, "round_robin"),
 				ExpectError: regexache.MustCompile("Stickiness type 'source_ip' is not supported for target groups with"),
 			},
 			{
-				Config:      testAccTargetGroupConfig_stickinessValidity(rName, "HTTPS", "source_ip", true),
+				Config:      testAccTargetGroupConfig_stickinessValidity(rName, "HTTPS", "source_ip", true, "round_robin"),
 				ExpectError: regexache.MustCompile("Stickiness type 'source_ip' is not supported for target groups with"),
 			},
 			{
-				Config:      testAccTargetGroupConfig_stickinessValidity(rName, "TLS", "lb_cookie", true),
-				ExpectError: regexache.MustCompile("Stickiness type 'lb_cookie' is not supported for target groups with"),
+				Config:      testAccTargetGroupConfig_stickinessValidity(rName, "HTTP", "lb_cookie", true, "weighted_random"),
+				ExpectError: regexache.MustCompile("You cannot have both stickiness and weighted random algorithm enabled on the same target group."),
+			},
+			{
+				Config:      testAccTargetGroupConfig_stickinessValidity(rName, "HTTPS", "lb_cookie", true, "weighted_random"),
+				ExpectError: regexache.MustCompile("You cannot have both stickiness and weighted random algorithm enabled on the same target group."),
 			},
 			{
-				Config:             testAccTargetGroupConfig_stickinessValidity(rName, "TCP_UDP", "lb_cookie", false),
+				Config:      testAccTargetGroupConfig_stickinessValidity(rName, "TLS", "lb_cookie", true, "round_robin"),
+				ExpectError: regexache.MustCompile("You cannot enable stickiness on target groups with the TLS protocol"),
```

Comment:
When `load_balancer_algorithm_type` was added to the resource it must have engaged a change in the precedence of checks on the AWS api side which caused a change to the error check here.

Per https://docs.aws.amazon.com/elasticloadbalancing/latest/network/load-balancer-target-groups.html#sticky-sessions
> Sticky sessions are not supported for TLS listeners.